writers/influxdb/messages.go

```diff
@@ -22,41 +23,80 @@ const pointName = "messages"
 var _ writers.MessageRepository = (*influxRepo)(nil)
 
 type influxRepo struct {
-	database string
-	client   influxdata.Client
+	client    influxdata.Client
+	batch     []*influxdata.Point
+	batchSize int
+	mu        sync.Mutex
+	tick      <-chan time.Time
+	cfg       influxdata.BatchPointsConfig
 }
 
 type fields map[string]interface{}
 type tags map[string]string
 
 // New returns new InfluxDB writer.
-func New(client influxdata.Client, database string) (writers.MessageRepository, error) {
-	return &influxRepo{database, client}, nil
+func New(client influxdata.Client, database string, batchSize int, batchTimeout time.Duration) (writers.MessageRepository, error) {
+	repo := &influxRepo{
+		client: client,
+		cfg: influxdata.BatchPointsConfig{
+			Database: database,
+		},
+		batchSize: batchSize,
+		batch:     []*influxdata.Point{},
+	}
+
+	repo.tick = time.NewTicker(batchTimeout).C
+	go func() {
+		for {
+			<-repo.tick
+			repo.save()
+		}
+	}()
+
+	return repo, nil
 }
 
-func (repo *influxRepo) Save(msg mainflux.Message) error {
-	bp, err := influxdata.NewBatchPoints(influxdata.BatchPointsConfig{
-		Database: repo.database,
-	})
+func (repo *influxRepo) save() error {
+	repo.mu.Lock()
+	defer repo.mu.Unlock()
+	bp, err := influxdata.NewBatchPoints(repo.cfg)
 	if err != nil {
 		return err
 	}
 
+	bp.AddPoints(repo.batch)
+
+	if err := repo.client.Write(bp); err != nil {
+		return err
+	}
+
+	repo.batch = []*influxdata.Point{}
+	return nil
```

Comment:
Should not this one reset the timeout timer also? I.e. if we had more writes than a batchsize we should write immediately, and then rest the timeout to start collecting a batch for a next second.